File: graffiti/src/main/java/com/ue/graffiti/util/SPUtils.java
package com.ue.graffiti.util;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;
import android.preference.PreferenceManager;

public final class SPUtils {

    private static SharedPreferences sharedPreferences;

    private SPUtils() {
        throw new UnsupportedOperationException();
    }

    public static void init(Application application) {
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(application);
    }

    public static void putString(String key, String value) {
        Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.commit();
    }

    public static void putInt(String key, int value) {
        Editor editor = sharedPreferences.edit();
        editor.putInt(key, value);
        editor.commit();
    }

    public static void putBoolean(String key, boolean value) {
        Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.commit();
    }

    public static void putLong(String key, long value) {
        Editor editor = sharedPreferences.edit();
        editor.putLong(key, value);
        editor.commit();
    }

    public static String getString(String key, String defaultVaule) {
        return sharedPreferences.getString(key, defaultVaule);
    }

    public static int getInt(String key, int defaultValue) {
        return sharedPreferences.getInt(key, defaultValue);
    }

    public static Boolean getBoolean(String key, boolean defaultValue) {
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public static long getLong(String key, long defaultValue) {
        return sharedPreferences.getLong(key, defaultValue);
    }

    public static void remove(String key) {
        Editor editor = sharedPreferences.edit();
        editor.remove(key);
        editor.commit();
    }

    public static void clear(Context ctx) {
        Editor editor = sharedPreferences.edit();
        editor.clear();
        editor.commit();
    }
}

File: library/src/main/java/com/ue/library/util/SPUtils.kt
package com.ue.library.util

import android.app.Application
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SPUtils private constructor() {

    init {
        throw UnsupportedOperationException()
    }

    companion object {

        private lateinit var sharedPreferences: SharedPreferences

        fun init(application: Application) {
            sharedPreferences = PreferenceManager.getDefaultSharedPreferences(application)
        }

        fun putString(key: String, value: String) = sharedPreferences.edit().putString(key, value).apply()

        fun putInt(key: String, value: Int) = sharedPreferences.edit().putInt(key, value).apply()

        fun putBoolean(key: String, value: Boolean) = sharedPreferences.edit().putBoolean(key, value).apply()

        fun putLong(key: String, value: Long) = sharedPreferences.edit().putLong(key, value).apply()

        fun getString(key: String, defaultValue: String = ""): String = sharedPreferences.getString(key, defaultValue)

        fun getInt(key: String, defaultValue: Int = 0) = sharedPreferences.getInt(key, defaultValue)

        fun getBoolean(key: String, defaultValue: Boolean = true) = sharedPreferences.getBoolean(key, defaultValue)

        fun getLong(key: String, defaultValue: Long = 0) = sharedPreferences.getLong(key, defaultValue)

        fun remove(key: String) = sharedPreferences.edit().remove(key).apply()

        fun clear() = sharedPreferences.edit().clear().apply()
    }
}